homeassistant/components/nws/weather.py

```diff
@@ -0,0 +1,306 @@
+    prec_prob = [w[1] for w in weather]
+
+    # Choose condition with highest priority.
+    cond = next((key for key, value in CONDITION_CLASSES.items()
+                 if any(condition in value for condition in conditions)),
+                conditions[0])
+
+    if cond == 'clear':
+        if time == 'day':
+            return 'sunny', max(prec_prob)
+        if time == 'night':
+            return 'clear-night', max(prec_prob)
+    return cond, max(prec_prob)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the nws platform."""
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    station = config.get(CONF_STATION)
+    api_key = config.get(CONF_API_KEY)
+
+    if None in (latitude, longitude):
+        _LOGGER.error("Latitude/longitude not set in Home Assistant config")
+        return
+
+    from pynws import Nws
+
+    websession = async_get_clientsession(hass)
+    # ID request as being from HA, pynws prepends the api_key in addition
+    api_key_ha = [api_key + 'homeassistant']
+    nws = Nws(websession, latlon=(float(latitude), float(longitude)),
+              userid=api_key_ha)
+
+    _LOGGER.debug("Setting up station: %s", station)
+    if station == '':
+        with async_timeout.timeout(10, loop=hass.loop):
+            stations = await nws.stations()
+        _LOGGER.info("Station list: %s", stations)
+        nws.station = stations[0]
+        _LOGGER.debug("Initialized for coordinates %s, %s -> station %s",
+                      latitude, longitude, stations[0])
+    else:
+        nws.station = station
+        _LOGGER.debug("Initialized station %s", station[0])
+
+    async_add_entities([NWSWeather(nws, hass.config.units, config)], True)
+
+
+class NWSWeather(WeatherEntity):
+    """Representation of a weather condition."""
+
+    def __init__(self, nws, units, config):
+        """Initialise the platform with a data instance and station name."""
+        self._nws = nws
+        self._station_name = config.get(CONF_NAME, self._nws.station)
+        self._observation = None
+        self._forecast = None
+        self._description = None
+        self._is_metric = units.is_metric
+        self._mode = config[CONF_MODE]
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self):
+        """Update Condition."""
+        with async_timeout.timeout(10, loop=self.hass.loop):
+            _LOGGER.debug("Updating station observations %s",
+                          self._nws.station)
+            self._observation = await self._nws.observations()
+            _LOGGER.debug("Updating forecast")
+            if self._mode == 'daynight':
+                self._forecast = await self._nws.forecast()
+            elif self._mode == 'hourly':
+                self._forecast = await self._nws.forecast_hourly()
+            else:
```

Comment:
Will never happen as the configuration needs to pass the validation before the integration is set up.